api/v1beta1/module_types.go

```diff
@@ -63,12 +52,8 @@ type Build struct {
 	Dockerfile string `json:"dockerfile"`
 
 	// +optional
-	// Pull contains settings determining how to pull the base images of the build process.
-	Pull PullOptions `json:"pull"`
-
-	// +optional
-	// Push contains settings determining how to push a built DriverContainer image.
-	Push PushOptions `json:"push"`
+	// BaseImageRegistryTLS contains settings determining how to pull the base images of the build-process' Dockerfile.
```

Comment:
why is it used only for pulling image, but not for pushing? You have pushing in the KernelMapping, but not here. Prior to your change it was on the opposite. What am i missing?

Reply:
This is one of the reason that PR was created. We don't need both.

We are always pushing to the same image registry as the one we are trying to pull the image from to begin with (we are building if the image doesn't exist), therefore, there is no reason to duplicate those configs. We should take them from the kernelMapping instead.

Reply:
Why the TLS, which is needed only in build, is put outside of the build? it is not a very clear API.

Reply:
It is not only needed for build, it is needed for access that image registry, if the image doesn't exist then we will build it and push it but we still need those configs for checking if the image exists prior to the build.